scarches/models/scgen/vaearith_model.py

```diff
@@ -0,0 +1,409 @@
+        """
+        device = next(self.model.parameters()).device
+        data = torch.tensor(data, device=device) # to tensor
+        rec_data = self.model.reconstruct(data, use_data)
+        rec_data = rec_data.cpu().detach()
+        return np.array(rec_data)
+
+
+    def predict(self, adata, conditions, cell_type_key, condition_key, adata_to_predict=None, celltype_to_predict=None, obs_key="all"):
+        """
+        Predicts the cell type provided by the user in stimulated condition.
+
+        Parameters
+        ----------
+        celltype_to_predict: basestring
+            The cell type you want to be predicted.
+        obs_key: basestring or dict
+            Dictionary of celltypes you want to be observed for prediction.
+        adata_to_predict: `~anndata.AnnData`
+            Adata for unpertubed cells you want to be predicted.
+
+        Returns
+        -------
+        predicted_cells: numpy nd-array
+            `numpy nd-array` of predicted cells in primary space.
+        delta: float
+            Difference between stimulated and control cells in latent space
+        """
+        #device = next(self.model.parameters()).device # get device of model.parameters
+        #adata_tensor = torch.tensor(adata, device=device) # to tensor
+        output = self.model.predict(adata, conditions, cell_type_key, condition_key, adata_to_predict, celltype_to_predict, obs_key)
+        prediction = output[0].cpu().detach()
+        delta = output[1].cpu().detach()
+        return np.array(prediction), np.array(delta)
+
+
+    def batch_removal(self, adata, batch_key, cell_label_key, return_latent = True):
+        """
+        Removes batch effect of adata
+
+        Parameters
+        ----------
+        adata: `~anndata.AnnData`
+            Annotated data matrix. adata must have `batch_key` and `cell_label_key` which you pass to the function in its obs.
+        batch_key: `str`
+            batch label key in  adata.obs
+        cell_label_key: `str`
+            cell type label key in adata.obs
+        return_latent: `bool`
+            if `True` returns corrected latent representation
+
+        Returns
+        -------
+        corrected: `~anndata.AnnData`
+            adata of corrected gene expression in adata.X and corrected latent space in adata.obsm["corrected"].
+        """
+        corrected = self.model.batch_removal(adata, batch_key, cell_label_key, return_latent)
+        return corrected
+
+
+    @classmethod
+    def _validate_adata(cls, adata, dct):
+        if adata.n_vars != dct['x_dim_']:
+            raise ValueError("Incorrect var dimension")
+
+    @classmethod
+    def _get_init_params_from_dict(cls, dct):
+        init_params = {
+            'hidden_layer_sizes': dct['hidden_layer_sizes_'],
+            'z_dimension': dct['z_dim_'],
+            'dr_rate': dct['dr_rate_'],
+        }
+
+        return init_params
+
+    @classmethod
+    def load_query_data(cls,
```

Comment:
Again not needed here, what to expand and freeze if no conditional weights are present.